src/services/ArtifactService.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * Artifact Service
+ */
+
+const AWS = require('aws-sdk')
+const config = require('config')
+const errors = require('common-errors')
+const fileTypeFinder = require('file-type')
+const joi = require('joi')
+const path = require('path')
+const _ = require('lodash')
+const s3 = new AWS.S3()
+const logger = require('../common/logger')
+const HelperService = require('./HelperService')
+
+/*
+ * Function to upload file to S3
+ * @param {Object} file File to be uploaded
+ * @param {String} name File name
+ * @return {Promise}
+ **/
+function * _uploadToS3 (file, name) {
+  const params = {
+    Bucket: config.aws.ARTIFACT_BUCKET,
+    Key: name,
+    Body: file.data,
+    ContentType: file.mimetype,
+    Metadata: {
+      originalname: file.name
+    }
+  }
+  // Upload to S3
+  return s3.upload(params).promise()
+}
+
+/**
+ * Function to download Artifact from S3
+ * @param {String} submissionId Submission ID
+ * @param {String} fileName File name which need to be downloaded from S3
+ * @return {Object} File downloaded from S3
+ */
+function * downloadArtifact (submissionId, fileName) {
+  const artifacts = yield s3.listObjects({Bucket: config.aws.ARTIFACT_BUCKET, Prefix: `${submissionId}/${fileName}`}).promise()
+  if (artifacts.Contents.length === 0) {
+    throw new errors.HttpStatusError(400, `Artifact ${fileName} doesn't exist for ${submissionId}`)
+  }
+  const key = artifacts.Contents[0].Key
+  const downloadedFile = yield s3.getObject({ Bucket: config.aws.ARTIFACT_BUCKET, Key: key }).promise()
+  // Return the retrieved Artifact
+  logger.info(`downloadArtifact: returning artifact ${fileName} for Submission ID: ${submissionId}`)
+  return { fileName: key.substring(key.lastIndexOf('/') + 1), file: downloadedFile.Body }
+}
+
+downloadArtifact.schema = {
+  submissionId: joi.string().uuid().required(),
+  fileName: joi.string().trim().required()
+}
+
+/**
+ * Function to list Artifacts present in S3 bucket
+ * @param {String} submissionId Submission ID
+ * @return {Object} List of files present in S3 bucket under submissionId directory
+ */
+function * listArtifacts (submissionId) {
+  const artifacts = yield s3.listObjects({Bucket: config.aws.ARTIFACT_BUCKET, Prefix: submissionId}).promise()
```

Comment:
This one does not verify first if the submission for the submission id passed seems to exist or not - I passed a random guid and it returned an empty array; was expecting an error.

Reply:
should return an error on a bad GUID for submission